server/base/src/main/java/org/apache/accumulo/server/client/ClientServiceHandler.java

```diff
@@ -216,12 +216,11 @@ public void grantTablePermission(TInfo tinfo, TCredentials credentials, String u
     NamespaceId namespaceId;
     try {
       namespaceId = context.getNamespaceId(tableId);
+      security.grantTablePermission(credentials, user, tableId, tableName,
+          TablePermission.getPermissionById(permission), namespaceId);
     } catch (TableNotFoundException e) {
       throw new TException(e);
     }
-
-    security.grantTablePermission(credentials, user, tableId,
```

Comment:
Was this moved because it now throws TableNotFoundException?

Reply:
Yes